### PR TITLE
Updating ExtensionStartupRunnerGenerator to fix the namespace conflict with customer code

### LINKED ISSUE #2542: ExtensionStartupRunnerGenerator generating code which conflicts with customer code (namespace)
### Description

The `ExtensionStartupRunnerGenerator` source generator is generating code which conflicts with customer code.

### Steps to reproduce

1. Create a function app with a service bus trigger.
2. Create a new class in the project with below code.

```csharp
namespace TheProjectNameFromStep1.Exception
{
    internal class Foo 
    {
    }
}
```
_Replace `TheProjectNameFromStep1` with your project name._

This will cause `CS0118	'Exception' is a namespace but is used like a type` error because our source generator created the below code where `Exception` type is used in the `catch` block.

```csharp
public override void Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)
{
    try
    {
        new global::Microsoft.Azure.Functions.Worker.ServiceBusExtensionStartup().Configure(applicationBuilder);
    }
    catch (Exception ex)
    {
        // removed 
    }
}
```

Additionally, double-check other function source generators to see if they have the same issue.

### PR DESCRIPTION
**Pull request by surgupta-msft**
### Issue describing the changes in this PR

resolves #2542 

This PR addresses the namespace conflict issue in `ExtensionStartupRunnerGenerator`. There will be follow-up PRs to address the issue in other generators.

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [X] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [X] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs [hunk 1]
```diff
@@ -216,7 +216,7 @@ private static string GenerateStartupCodeExecutorClass(IList<string> startupType
                                     {
                                         new global::{{typeName}}().Configure(applicationBuilder);
                                     }
-                                    catch (Exception ex)
+                                    catch (global::System.Exception ex)
```
Comment (kshyju) [inline sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs:219]: Any other generators need a similar fix?

Comment (surgupta-msft) [reply sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs:219]: I don't think so. But let me know if you want me to check anything specific.

**Comment by jviau** [comment]
This isn't the first time we have had to add `global::` to our generator code. Can we do a pass to see if any other parts need it?

File: sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs (hunk 1 above)
Comment (surgupta-msft) [reply sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs:219]: We can keep this PR scoped to `ExtensionStartupRunnerGenerator` and I will prepare followup PRs for other generators.

**Review by kshyju — APPROVED** [review]

**Pull request merged**